python/ray/tune/tests/test_api.py

```diff
@@ -1465,6 +1471,42 @@ def testCreateScheduler(self):
         real_scheduler = AsyncHyperBandScheduler(**kwargs)
         assert type(shim_scheduler) is type(real_scheduler)
 
+    def testCreateAllSchedulers(self):
+        # key word argument template
+        kwargs = {"metric": "metric_foo", "mode": "min"}
+
+        # get list of all schedulers to test
+        schedulers_to_test = [scheduler for scheduler in SCHEDULER_IMPORT.keys()]
+
+        for scheduler_id in schedulers_to_test:
+            kwargs_for_test = kwargs.copy()
+            if scheduler_id == "pbt_replay":
+                # using /dev/null for policy file placeholder for this test
+                # assumes existence of policy file is sufficient for this test
+                # and not the actual content
+                kwargs_for_test = {"policy_file": "/dev/null"}
+            elif scheduler_id == "pb2":
+                # add scheduler required parameter
+                kwargs_for_test["hyperparam_bounds"] = {"param1": [0, 1]}
+
+            shim_scheduler = tune.create_scheduler(scheduler_id, **kwargs_for_test)
+
+            # create scheduler with actual class
+            SchedulerClass = SCHEDULER_IMPORT[scheduler_id]
+            if inspect.isfunction(SchedulerClass):
+                # handle case of wrapping function
+                SchedulerClass = SchedulerClass()
```

Comment:
I feel like this slightly defeats the purpose of the test, since it's just executing a copy of the code that's being tested.

Maybe we can change this to simply test this new function pattern, and confirm that calling this function on "pb2" will properly return a `PB2` object.

Reply:
@matthewdeng   Not sure I completely understand the comment.  I made a guess at what you meant.  Let me know if the change is what you are looking for.  For now I'm leaving this as unresolved.

Reply:
Ah I was thinking of something even more simple, like:

``` python
def testCreateLazyImportSearcher(self):
    kwargs = {"metric": "metric_foo", "mode": "min", "hyperparam_bounds": {"param1": [0, 1]}}
    shim_searcher_pb2 = tune.create_searcher("pb2", **kwargs)
    real_searcher_pb2 = PB2(**kwargs)
    assert type(shim_searcher_pb2) is type(real_searcher_pb2)
```

Reply:
@matthewdeng   As I understand the guidance, I replaced `testCreateAllSchedures` with the `testCreateLazyScheduler` test.  Here is a run with the revised test
```
root@e9f92801c6b4:/opt/project# pytest -v /opt/project/python/ray/tune/tests/test_api.py::ShimCreationTest
Test session starts (platform: linux, Python 3.8.13, pytest 5.4.3, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /opt/project/python
plugins: lazy-fixture-0.6.3, rerunfailures-10.2, anyio-3.6.1, sugar-0.9.4, asyncio-0.16.0, virtualenv-1.7.0, shutil-1.7.0, typeguard-2.13.3, remotedata-0.3.3, timeout-2.1.0
collecting ...
 ray/tune/tests/test_api.py::ShimCreationTest.testCreateLazyImportScheduler ✓                                                 25% ██▌
 ray/tune/tests/test_api.py::ShimCreationTest.testCreateScheduler ✓                                                           50% █████
 ray/tune/tests/test_api.py::ShimCreationTest.testCreateSearcher ✓                                                            75% ███████▌
 ray/tune/tests/test_api.py::ShimCreationTest.testExtraParams ✓                                                              100% ██████████
============================================================= warnings summary =============================================================
```

One last note, after pushing up a the changes, I noticed the CI still seems to indicate lint issues.  In looking at the output, I can seem to errors related to this PR.  The errors I see are with modules that this PR does not change.  

As a sanity check, this is the output from `scripts/format.sh` on my local dev environment:
```
root@e9f92801c6b4:/opt/project# ./scripts/format.sh
INFO: Ray uses shellcheck for shell scripts, which is not installed. You may install shellcheck=0.7.1 with your system package manager.
WARNING: clang-format is not installed!
WARNING:java is not installed, skip format java files!
From https://github.com/ray-project/ray
 * branch                master     -> FETCH_HEAD
All done! ✨ 🍰 ✨
2 files left unchanged.
root@e9f92801c6b4:/opt/project#
```

If there are BK CI still call out a lint issue, I'll need your help to identifying what is the problem.

Let me know if this is the change you were looking for.

Reply:
Everything looks good!